### ISSUE TITLE
HEAL study 870 is not pulling over from the TIN

### ISSUE THREAD

**Issue by dcarsey** [RENCI/ctmd#484]
HEAL study 870 is not pulling over from the TIN.  

I thought that it needed a submission date to be pulled over, but the other HEAL studies do not have submission dates either.

**Comment by krobasky** [comment]
@hyi says:
I can confirm that this HEAL study 870 is indeed pulled over from the readcap TIN correctly, but it looks like the filtering done on our side filtered out study 870 so it did not get added to the database. We suspect data filtering could be most likely done here in this scala file: https://github.com/RENCI/map-pipeline/blob/97a42a6bc1934d72c8d5bd2c39535e4df8600930/src/main/scala/tic/Transform2.scala. 

@jdr0887 is looking into it, borrowing his LoE from ImmCellFIE (and we are VERY grateful for his help)

**Comment by krobasky** [comment]
Here's the KLUDGE we devised today for a quick fix to get Duke running again:
 - [ ] @hyi will create a 'ctmd' service account on stage-heal-ctmd.edc.renci.org
 - [ ] @rajshruti18  will make it her top priority to install ctmd, soup-to-nuts, mirroring heal-ctmd.edc.renci.orgI'll help if I can, this will include copying over the .env and other config files, synch'ing, and testing to be sure the servers are "identical"
 - [ ] @hyi will edit the .env file to remove nightly synch and manually add the new proposal to the database on stage, and check if that works; if it does, then when @dcarsey accepts the release, @rajshruti18 will deploy the same solution to heal-ctmd for the quick fix.

WARNING: this is extremely KLUDGEY and unsustainable

It buys us time: if this works, we might have a solution for heal-ctmd to make it usable very quickly.

We're still going to need to spend some SERIOUS time to address sustainability of the pipeline code.

**Comment by krobasky** [comment]
Speedbump : stage-heal-ctmd.edc.renci.org is behind the RENCI firewall and also isn't proxied through the Kemp, so the Vandy authenticator can't call back to stage-heal-ctmd.renci.org, which would be in front of the firewall if it existed. I've asked RCIADMIN if they can assist, urgently.
Also, stage-heal-ctmd isn't working on https, I'm not sure if that's going to be a problem.

**Comment by jdr0887** [comment]
Ok, I have a hunch.  I think the "pi_firstname" for proposal_id 870 is failing the DSL.NameParser.isWellFormedFirstName method from https://github.com/RENCI/map-pipeline/blob/97a42a6bc1934d72c8d5bd2c39535e4df8600930/src/main/scala/tic/Transform2.scala#L64

The pi_firstname field has an honorific that causes the function to return false ("Dr. ").  No other pi_firstname has an honorific.

**Comment by dcarsey** [comment]
Wowwwww. We have had problems with a poorly formed name once before when they tried to shove two first names and two last names into the pi name?

I’ll let the rest of you respond to this.

Thanks for these clever thoughts, Jason!

D
________________________________

**Comment by hyi** [comment]
I compared the log from the ctmd-pipeline between prod and stage and found after running this left join filtering: https://github.com/RENCI/map-pipeline/blob/97a42a6bc1934d72c8d5bd2c39535e4df8600930/src/main/scala/tic/Transform2.scala#L147, proposal 870 is filtered out on prod but kept without being filtered out on stage. So I tried to manually insert proposal 870 on prod into the Proposal table using the proposal 870 data on stage, and it looks good after being inserted, however, after nightly data sync, the proposal 870 was wiped out by the nightly data sync. On stage, there is no initial data initially and nightly data sync pulled data correctly including proposal 870, and afterwards nightly data sync never wiped out proposal 870 on stage, but somehow wiped out on prod. Not sure what other data on prod could cause this different behavior on stage. We have restored prod data on stage, and will see whether data sync tonight will wipe out proposal 870 tomorrow. If that happens, at least we can reproduce the issue on stage. Currently this problem only happens on prod and we haven't been able to reproduce it on stage yet.
